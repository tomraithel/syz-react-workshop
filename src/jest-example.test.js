import React from "react";
import TestRenderer from "react-test-renderer";

const HappyIndicator = ({ isHappy = true }) => (
  <div>{isHappy ? "😄" : "🙁"}</div>
);

test("Renders happy", () => {
  const result = TestRenderer.create(<HappyIndicator />);
  expect(result).toMatchSnapshot();
});

test("Renders sad", () => {
  const result = TestRenderer.create(<HappyIndicator isHappy={false} />);
  expect(result).toMatchSnapshot();
});

test("Renders sad v2", () => {
  const result = TestRenderer.create(<HappyIndicator isHappy={false} />);
  expect(result.root.findByType("div").props.children).toBe("🙁");
});

const sum = (a, b) => a + b;

test("Should sum 1 + 2 correctly", () => {
  expect(sum(1, 2)).toBe(3);
});
