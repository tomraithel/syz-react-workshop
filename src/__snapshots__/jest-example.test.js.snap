// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders happy 1`] = `
<div>
  😄
</div>
`;

exports[`Renders sad 1`] = `
<div>
  🙁
</div>
`;
